src/derivations/Segmentation.js

```diff
@@ -0,0 +1,278 @@
+            BitsAllocated: "1",
+            BitsStored: "1",
+            HighBit: "0",
+            PixelRepresentation: "0",
+            LossyImageCompression: "00",
+            SegmentationType: "BINARY",
+            ContentLabel: "EXAMPLE"
+        });
+
+        let dimensionUID = DicomMetaDictionary.uid();
+        this.dataset.DimensionOrganizationSequence = {
+            DimensionOrganizationUID: dimensionUID
+        };
+        this.dataset.DimensionIndexSequence = [
+            {
+                DimensionOrganizationUID: dimensionUID,
+                DimensionIndexPointer: 6422539,
+                FunctionalGroupPointer: 6422538, // SegmentIdentificationSequence
+                DimensionDescriptionLabel: "ReferencedSegmentNumber"
+            },
+            {
+                DimensionOrganizationUID: dimensionUID,
+                DimensionIndexPointer: 2097202,
+                FunctionalGroupPointer: 2134291, // PlanePositionSequence
+                DimensionDescriptionLabel: "ImagePositionPatient"
+            }
+        ];
+
+        this.dataset.SegmentSequence = [];
+
+        // TODO: check logic here.
+        // If the referenced dataset itself references a series, then copy.
+        // Otherwise, reference the dataset itself.
+        // This should allow Slicer and others to get the correct original
+        // images when loading Legacy Converted Images, but it's a workaround
+        // that really doesn't belong here.
+        if (this.referencedDataset.ReferencedSeriesSequence) {
+            this.dataset.ReferencedSeriesSequence = DerivedDataset.copyDataset(
+                this.referencedDataset.ReferencedSeriesSequence
+            );
+        } else {
+            const ReferencedInstanceSequence = [];
+
+            for (let i = 0; i < this.referencedDatasets.length; i++) {
+                ReferencedInstanceSequence.push({
+                    ReferencedSOPClassUID: this.referencedDatasets[i]
+                        .SOPClassUID,
+                    ReferencedSOPInstanceUID: this.referencedDatasets[i]
+                        .SOPInstanceUID
+                });
+            }
+
+            this.dataset.ReferencedSeriesSequence = {
+                SeriesInstanceUID: this.referencedDataset.SeriesInstanceUID,
+                StudyInstanceUID: this.referencedDataset.StudyInstanceUID,
+                ReferencedInstanceSequence
+            };
+        }
+
+        if (!this.options.includeSliceSpacing) {
+            // per dciodvfy this should not be included, but dcmqi/Slicer requires it
+            delete this.dataset.SharedFunctionalGroupsSequence
+                .PixelMeasuresSequence.SpacingBetweenSlices;
+        }
+
+        // make an array of zeros for the pixels assuming bit packing (one bit per short)
+        // TODO: handle different packing and non-multiple of 8/16 rows and columns
+        // The pixelData array needs to be defined once you know how many frames you'll have.
```

Comment:
We have some "weird" synthetic datasets that can be used to test this aspect of implementation, if you get to doing that.

Reply:
We definitely need to set up the regression tests with a set of example datasets that cover as wide a spectrum of "valid" cases as possible, together with any real-world data that we want to support even if it's not strictly standard.